validators/cgroup_validator.go

```diff
@@ -82,6 +85,31 @@ func (c *CgroupsValidator) validateCgroupSubsystems(cgroups, subsystems []string
 }
 
 func (c *CgroupsValidator) getCgroupSubsystems() ([]string, error) {
+	// Get the subsystems from /sys/fs/cgroup/cgroup.controllers when cgroup v2 is used.
+	// /proc/cgroups is meaningless for v2
+	// https://github.com/torvalds/linux/blob/v5.3/Documentation/admin-guide/cgroup-v2.rst#deprecated-v1-core-features
+	var st unix.Statfs_t
+	if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
+		return []string{}, errors.Wrap(err, "cannot statfs the cgroupv2 root")
+	}
+	if st.Type == unix.CGROUP2_SUPER_MAGIC {
```

Comment:
Maybe split these cases to make them easier to test?

Reply:
Fixed it